### PR TITLE
Allow handler error with an express middleware

### PR DESCRIPTION
**Pull request by adriexnet**
### What does this PR do?
Allow using an express middleware as handler error

### How should this be tested?
• Configure graphqlHTTP with handleErrorWithExpress
```
app.use('/graph', graphqlHTTP(async (req) => {
  return {
    graphiql: true,
    schema,
    handleErrorWithExpress: true,
    formatError(error) {
      logger.error(error.message, {
        path: error.path,
        locations: error.locations,
        stack: error.stack ? error.stack.split('\n') : [],
        trace: error.trace,
      });
      return {
        message: error.message,
        status: 400,
      };
    }
  };
}));
```
• Add an error middleware
```
app.use((error, req, res, next) => {
  if (res.headersSent) {
    return next(error);
  }

  return res.status(error.status || 500).send({
    error: error.message
  });
});
```

### CODE REVIEW

**Comment by ValfarDeveloper** [comment]
Oh man this is so useful! It's a shame that yet is not implemented.

File: src/index.js [hunk 1]
```diff
@@ -336,6 +350,16 @@ function graphqlHTTP(options: Options): Middleware {
...
         }
 
+        // If handler error with express is allowed, return the first error in next function.
+        if (
+          handleErrorWithExpress &&
+          result &&
+          result.errors &&
+          result.errors.length
+        ) {
+          return nextFunction(result.errors[0]);
```
Comment (jsonmaur) [inline src/index.js]: I'd say we should return all the errors in the array, not just the first one.

**Comment by aagtt** [comment]
for god sake, accept this!

**Comment by driescroons** [comment]
@aagtt @jsonmaur 

I wouldn't add the ability to add error middleware to the package itself. I've handled it like so:
- I use express-async-errors (https://www.npmjs.com/package/express-async-errors)
- I throw the error in my customFormatErrorFn
- My error middleware catches the error and is able to handle it according to the type

Here a demo:
https://streamable.com/hvtjg3

**Pull request closed without merging**